feat: list common disconnect reasons (#3111)
emqtt/emq-docs-cn: en_US/admin/cli.md

  max_conns       : 5000000
```

#### Common shutdown reasons

| Reason                        | Description                                                                                           |
|-------------------------------|-------------------------------------------------------------------------------------------------------|
| `banned`                      | The client is blacklisted due to ACL violations, rate limiting, or IP restrictions.                   |
| `closed`                      | The connection was closed either by the server or the client.                                         |
| `discarded`                   | A client with the same clientid connected with `clean_start = true` while the old session was alive.  |
| `takenover`                   | A client with the same clientid connected with `clean_start = false` while the old session was alive. |

> | `einval`                      | Invalid argument or socket error, typically indicating a network or client SDK issue.                 |

Reviewer: ```suggestion
| `einval`                      | Invalid argument or socket error, happens when EMQX tries to write data to a socket which is closed already by the system (a race condition between socket state change notification and data write).                 |
```

| `frame_too_large`             | The MQTT packet exceeds the maximum allowed frame size.                                               |

> | `idle_timeout`                | The connection was idle beyond the configured timeout period.                                         |

Reviewer: ```suggestion
| `idle_timeout`                | The `CONNECT` packet is not received for too long after the configured timeout after connection is established.                                         |
```

| `invalid_proto_name`          | The protocol name in the `CONNECT` packet is invalid or not `"MQTT"`.                                 |
| `invalid_topic`               | The client used an invalid topic (e.g., containing illegal characters or forbidden by the broker).    |
| `keepalive_timeout`           | The client failed to send any packets within the keepalive interval.                                  |
| `malformed_packet`            | The MQTT packet is corrupted or does not conform to the MQTT specification.                           |
| `not_authorized`              | The client attempted an unauthorized action, rejected by ACL.                                         |
| `ssl_closed`                  | The SSL/TLS connection was closed by the peer.                                                        |
| `ssl_error`                   | A SSL/TLS handshake or data transmission error occurred.                                              |

> | `ssl_upgrade_timeout`         | Timeout during SSL/TLS upgrade or handshake.                                                          |

Reviewer: ```suggestion
| `ssl_upgrade_timeout`         | Timeout during SSL/TLS handshake.                                                          |
```

| `unexpected_packet`           | The client sent a packet that was unexpected in the current connection state.                         |
| `zero_remaining_len`          | The packet has a zero remaining length field, which is invalid in most contexts.                      |
| `bad_username_or_password`    | The client's credentials failed authentication.                                                       |

> | `client_identifier_not_valid` | The client ID is invalid or disallowed.                                                               |

Reviewer: ```suggestion
| `client_identifier_not_valid` | The client ID is being registered by another connection.                                                               |
```

| `protocol_error`              | A generic MQTT protocol violation.                                                                    |
| `tcp_closed`                  | The TCP connection was closed by the client or network layer.                                         |

> | `timeout`                     | A general timeout occurred (e.g., during authentication, SSL handshake, etc.).                        |

Reviewer: Q: SSL handshake timeout will result in `ssl_upgrade_timeout`, or `timeout`?

Author: Probably `ssl_upgrade_timeout`

Author: ```suggestion
| `timeout`                     | A general timeout occurred (e.g., during authentication, etc.).                        |
```

| `invalid_proto_name`          | The protocol name in the `CONNECT` packet is invalid or not `"MQTT"`. |
| `invalid_topic`               | The client used an invalid topic (e.g., containing illegal characters or forbidden by the broker). |
| `keepalive_timeout`           | The client failed to send any packets within the keepalive interval. |
| `malformed_packet`            | The MQTT packet is corrupted or does not conform to the MQTT specification. |
| `not_authorized`              | The client attempted an unauthorized action, rejected by ACL. |
| `ssl_closed`                  | The SSL/TLS connection was closed by the peer.               |
| `ssl_error`                   | An error occurred during the SSL/TLS handshake or data transmission. |
| `ssl_upgrade_timeout`         | The SSL/TLS handshake did not complete within the allowed time. |
| `unexpected_packet`           | The client sent a packet that was unexpected in the current connection state. |
| `zero_remaining_len`          | The packet has a zero remaining length field, which is invalid in most contexts. |
| `bad_username_or_password`    | Authentication failed due to incorrect username or password. |

> | `client_identifier_not_valid` | The provided `clientid` is invalid or is already in use by another connection. |

Reviewer: ```suggestion
| `client_identifier_not_valid` | The provided `clientid` is invalid or locked by another client during login. |
```
